core/src/main/java/org/apache/iceberg/PartitionStatsHandler.java

```diff
@@ -115,6 +120,53 @@ public static Schema schema(StructType unifiedPartitionType) {
         LAST_UPDATED_SNAPSHOT_ID);
   }
 
+  /**
+   * Generates the partition stats file schema based on a combined partition type which considers
+   * all specs in a table. (For format version 3 and above)
+   *
+   * @param unifiedPartitionType unified partition schema type. Could be calculated by {@link
+   *     Partitioning#partitionType(Table)}.
+   * @return a schema that corresponds to the provided unified partition type.
+   */
+  public static Schema schemaV3Plus(StructType unifiedPartitionType) {
```

Comment:
Could this be `public static Schema schema(StructType unifiedPartitionType, FormatVersion version) {`